aspnetcore/blazor/state-management/prerendered-state-persistence.md

```diff
@@ -0,0 +1,355 @@
+---
+title: ASP.NET Core Blazor prerendered state persistence
+author: guardrex
+description: Learn how to persist user data (state) in Blazor apps using Blazor's Persistent Component State service.
+monikerRange: '>= aspnetcore-8.0'
+ms.author: wpickett
+ms.custom: mvc
+ms.date: 08/05/2025
+uid: blazor/state-management/prerendered-state-persistence
+---
+# ASP.NET Core Blazor prerendered state persistence
+
+Without persisting component state, state used during prerendering is lost and must be recreated when the app is fully loaded. If any state is created asynchronously, the UI may flicker as the prerendered UI is replaced when the component is rerendered.
+
+Consider the following `PrerenderedCounter1` counter component. The component sets an initial random counter value during prerendering in [`OnInitialized` lifecycle method](xref:blazor/components/lifecycle#component-initialization-oninitializedasync). When the component then renders interactively, the initial count value is replaced when `OnInitialized` executes a second time.
+
+`PrerenderedCounter1.razor`:
+
+:::code language="razor" source="~/../blazor-samples/8.0/BlazorSample_BlazorWebApp/Components/Pages/PrerenderedCounter1.razor":::
+
+Run the app and inspect logging from the component. The following is example output.
+
+> [!NOTE]
+> If the app adopts [interactive routing](xref:blazor/fundamentals/routing#static-versus-interactive-routing) and the page is reached via an internal [enhanced navigation](xref:blazor/fundamentals/routing#enhanced-navigation-and-form-handling), prerendering doesn't occur. Therefore, you must perform a full page reload for the `PrerenderedCounter1` component to see the following output. For more information, see the [Interactive routing and prerendering](#interactive-routing-and-prerendering) section.
```

Comment:
The placement of this note makes it read like the note is the example output. Suggested new placement:

```suggestion
> [!NOTE]
> If the app adopts [interactive routing](xref:blazor/fundamentals/routing#static-versus-interactive-routing) and the page is reached via an internal [enhanced navigation](xref:blazor/fundamentals/routing#enhanced-navigation-and-form-handling), prerendering doesn't occur. Therefore, you must perform a full page reload for the `PrerenderedCounter1` component to see the following output. For more information, see the [Interactive routing and prerendering](#interactive-routing-and-prerendering) section.

Run the app and inspect logging from the component. The following is example output.
```